src/components/SettingsBox.js

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import ToggleSwitch, { ToggleSwitchName } from "./ToggleSwitch";
+import styled from "styled-components";
+
+const ModuleTitle = styled.h3`
+  color: #fff;
+  font-weight: 700;
+  margin-top: 10px;
+  margin-bottom: 10px;
+`;
+
+const SettingsBoxWrapper = styled.div`
+  width: 175px;
+  height: fit-content;
+  padding: 0px 10px 10px 10px;
+  background-color: transparent;
+  border: solid 5px #fff;
+  border-radius: 15px;
+`;
+
+const ToggleSwitchWrapper = styled.div`
+  width: 100%;
+  height: 30px;
+  display: flex;
+  align-items: center;
+`;
+
+const SettingsBox = ({ saveSettings, settings }) => {
+  const [settingsPanel, setSettingsPanel] = React.useState(settings);
```

Comment:
If the SettingsBox is controlled, you don't need an internal state here.
You can directly access the props.